src/Realm.C

```diff
@@ -497,6 +501,13 @@ Realm::initialize()
   timerPopulateFieldData_ += time;
   NaluEnv::self().naluOutputP0() << "Realm::ioBroker_->populate_field_data() End" << std::endl;
 
+  // rebalance mesh using stk_balance
+  if (rebalanceMesh_) {
```

Comment:
Shouldn't rebalance happen before `create_edges` (possibly?) but definitely before `populate_field_data`?

Reply:
The rebalance depends on the coordinates so it needs to happen after populate_field_data(), otherwise it will segfault.

Reply:
Interesting, I didn't know that. I was under the impression that Zoltan only required the connectivity graph. Do you know what it uses coordinates for? 

So the next question is how do we handle the interaction with `balance_nodes()` (called above in line 479). If the user activates both, then this Zoltan rebalance would undo what was done in `balance_nodes`, correct? So should we be setting `balance_nodes()` to happen after `balanceStkMesh`? I assume since only ownership is changing, fields are consistent and doesn't get affected?

Reply:
@sayerhs rebalance has a couple of different options for the algorithm that is used internally. If a graph-based algorithm is used, then coordinates would not be needed. But if a geometric algorithm is used (like RCB) then coordinates are needed. Also, stk-balance generally does a proximity search so that contact surfaces can be kept together in the decomposition, and the proximity search needs coordinates.

Good point about balance_nodes, that needs to happen after the rebalance. And yes the fields should be fine after rebalance, field-data gets moved with the entities that are being moved.

Ideally rebalance would be a little cheaper if it is done prior to create_edges. But that's less important than putting rebalance after populate_field_data (which is mandatory).